### DISCUSSION TITLE
Confusing ExternalForce function parameters (apply_force_at_point)

### DISCUSSION THREAD

**Discussion by Sergi-Ferrez** [avianphysics/avian#531] [Q&A]
I'm a little bit confused about the `apply_force_at_point` parameters, these are the parameters of the function:
```
pub fn apply_force_at_point(
        &mut self,
        force: Vector,  // 'world-space' force meaning global translation + force?
        point: Vector, // I think this is in local space?
        center_of_mass: Vector, // This needs to be in global or local space?
)
```
I think it's a little confusing for newcomers to try to understand, someone can help me to identify the correct parameter values? 
Thanks in advance.

**Comment by awest03** [comment]
Hi, I appreciate this is very old, but this is what I do:
```rust
ef.apply_force_at_point(
	            transform.affine().transform_vector3(self.force),
	            transform.affine().transform_vector3(self.pos),
	            Vec3::ZERO,
	        );
```
All of the parameters are supposed to be in local space, but it doesn't account for rotation, hence my transformation

**Reply by lucasw** [reply]
The documentation was [recently updated](https://github.com/Jondolf/avian/commit/114554bf61320e03d778c5d8ca67e9ae381476c5
)- it's world space, not local:

> The force, point, and center of mass must be given in world space.

https://github.com/Jondolf/avian/blob/main/src/dynamics/rigid_body/forces.rs#L159

I get a GlobalTransform for the object and [use `translation()` as the center of mass](https://github.com/lucasw/bevy_viz/blob/main/vehicle_avian/src/bin/vehicle.rs#L127) (though for some application maybe the center of mass should be shifted relative to that, but the center of the collider is probably a good starting point).
